dsgrid/utils/scratch_dir_context.py

```diff
@@ -0,0 +1,49 @@
+import os
+import shutil
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+
+class ScratchDirContext:
+    """Manages the lifetime of files in a scratch directory."""
+
+    def __init__(self, scratch_dir: Path):
+        self._scratch_dir = scratch_dir
+        self._paths: list[Path] = []
+        if not self._scratch_dir.exists():
+            self._scratch_dir.mkdir()
+
+    @property
+    def scratch_dir(self) -> Path:
+        """Return the scratch directory."""
+        return self._scratch_dir
+
+    def add_tracked_path(self, path: Path) -> None:
+        """Add tracking of a path in the scratch directory."""
+        self._paths.append(path)
+
+    def list_tracked_paths(self) -> list[Path]:
+        """Return a list of paths being tracked."""
+        return self._paths[:]
+
+    def get_temp_filename(self) -> Path:
+        """Return a temporary filename based in the scratch directory."""
+        with NamedTemporaryFile(dir=self._scratch_dir) as f:
+            return Path(f.name)
```

Comment:
How does this get closed/deleted? (Does it?)

Reply:
It is deleted as soon as this code block exits. It only serves to generate a unique name in the given path. The expectation is that callers are going to use the name. It would not work if we were doing multiprocessing, as there could be collisions (we are obviously not going to do that). I should note that.